8510Final-AR.qmd

```diff
@@ -0,0 +1,311 @@
+casebook.new <- casebook_final_2 %>%
+  mutate(date = as.Date(Full.Date, format = "%Y-%m-%d"),  
+         Year = year(date),
+         Month = month(date),
+         Day = day(date))
+custom_stop_words <- c("aberdeen", "weekly", "times", "journal", "article", "said", "mr", "b", "q.c", "c", "b.a", "w", "a", "i", "u", "m", "s", "t")
+head(casebook.new)
+
+```
+
+```{r}
+#3 I plan to filter the news articles into two separate datasets: one for articles published before 1888 and another for those published after. I will then combine the stop word lists for both datasets and remove any duplicates.
+
+pre_1888 <- casebook.new %>% filter(Year < 1888)
+post_1888 <- casebook.new %>% filter(Year >= 1888)
+
+data(stop_words)  
+custom_stop_words <- tibble(word = custom_stop_words, lexicon = rep("custom", length(custom_stop_words)))
+all_stop_words <- bind_rows(stop_words, custom_stop_words) %>%
+  distinct() 
+
+```
+
+Now I will code clean up my pre and post 1888 news reports by extracting individual words, removing common stop words, and filtering out short words from the rows. This leaves a dataset of cleaned words ready for topic modeling.
+
+```{r}
+#4 For pre-1888
+pre_1888_tidy <- pre_1888 %>%
+  unnest_tokens(word, News.Reports) %>%
+  anti_join(all_stop_words, by = "word") %>%
+  filter(str_length(word) > 1)  
+
+```
+
+```{r}
+#5 For post-1888
+post_1888_tidy <- post_1888 %>%
+  unnest_tokens(word, News.Reports) %>%
+  anti_join(all_stop_words, by = "word") %>%
+  filter(str_length(word) > 1)
+```
+
+Now I will use cast_dtm to turn the words into a dtm for each datset.
+
+```{r}
+#6 Creating the DTM for pre-1888
+dtm_pre_1888 <- pre_1888_tidy %>%
+  count(file_name = File.Names, word) %>%  
+  cast_dtm(document = file_name, term = word, value = n)
+
+```
+
+```{r}
+# 7 Creating the DTM for post-1888
+dtm_post_1888 <- post_1888_tidy %>%
+  count(file_name = File.Names, word) %>%  
+  cast_dtm(document = file_name, term = word, value = n)
+
+```
+
+```{r}
+#8 Applying LDA to the pre-1888 DTM
+lda_pre_1888 <- LDA(dtm_pre_1888, k = 8, control = list(seed = 1234))
+
+```
+
+```{r}
+#9 Applying LDA to the post-1888 DTM
+lda_post_1888 <- LDA(dtm_post_1888, k = 8, control = list(seed = 1234))
+```
+
+```{r}
+#10 Analyzing topics by extracting the terms for each topic and their probabilities
+library(broom)
+
+topics_pre_1888 <- tidy(lda_pre_1888, matrix = "beta")
+topics_post_1888 <- tidy(lda_post_1888, matrix = "beta")
+
+```
+
+```{r}
+#11 Filter topics for pre-1888
+top_terms_pre_1888 <- topics_pre_1888 %>%
+  filter(topic %in% 1:5) %>%
+  group_by(topic) %>%
+  top_n(5, beta) %>%
+  ungroup() %>%
+  arrange(topic, desc(beta))
+```
+
+```{r}
+#12 Filter topics for post-1888
+top_terms_post_1888 <- topics_post_1888 %>%
+  filter(topic %in% 1:5) %>%
+  group_by(topic) %>%
+  top_n(5, beta) %>%
+  ungroup() %>%
+  arrange(topic, desc(beta))
+
+```
+
+```{r}
+#13 Plotting for each
+library(ggplot2)
```

Comment:
You only need to load libraries once at the top. They should not be repeated throughout.